domain/application/resource/resourcestore.go

```diff
@@ -9,25 +9,35 @@ import (
 	"sync"
 
 	"github.com/juju/juju/core/objectstore"
+	coreresource "github.com/juju/juju/core/resources"
 	"github.com/juju/juju/domain/application/errors"
 	"github.com/juju/juju/internal/charm/resource"
 )
 
 // ResourceStore provides a list of methods necessary for interacting with
 // a store for the resource.
 type ResourceStore interface {
-	// Get returns an io.ReadCloser for data at path, namespaced to the
-	// model.
-	Get(context.Context, string) (io.ReadCloser, int64, error)
-}
+	// Get returns an io.ReadCloser for a resource in the resource store.
+	Get(
+		ctx context.Context,
+		resourceUUID coreresource.ID,
+	) (r io.ReadCloser, size int64, err error)
 
-// FileResourceStore is a ResourceStore for file resource types.
-type FileResourceStore struct {
-	objectStore objectstore.ObjectStore
-}
+	// Put stores data from io.Reader in the resource store at the
+	// path specified in the resource.
```

Comment:
Questions: What is the "path specified in the resource"? Are you talking about the resource table row? The resource metadata from the charm? Why would the storage path be specified before something is placed in it?

Reply:
Changed this to say "using the UUID as the key".